core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.ts

```diff
@@ -13,543 +13,641 @@
+  labelColor: string;
+  actionButtonTitle: string;
+}
+
+export interface TranslationContributionsSummaryList {
+  id: string;
+  heading: string;
+  subheading: string;
+  labelText: string;
+  labelColor: string;
+  actionButtonTitle: string;
+}
+
+export interface OpportunitiesDictsObject {
+  opportunitiesDicts: unknown;
+  more: unknown;
+}
+
+export interface ContributionDetails {
+  skill_description: string;
+  skill_rubrics: Rubric[];
+  chapter_title: string;
+  story_title: string;
+  topic_name: string;
+}
+
+export interface ContributionTabs {
+  suggestionType: string;
+  text: string;
+  enabled: boolean;
+}
+
+export interface SuggestionIdToSuggestions {
+  [key: string]: {
+    suggestion: Suggestion;
+    details: {
+      skill_description: string;
+      topic_name: string;
+      chapter_title: string;
+      story_title: string;
+    };
+  };
+}
+
+export interface SUGGESTION_LABELS {
+  [key: string]: {
+    text: string;
+    color: string;
+  };
+}
+
+export interface SuggestionContributions {
+  suggestion_type: string;
+  suggestion_id: string;
+  target_id: string;
+  change: {
+    content_html: string;
+    translation_html: string;
+  };
+  status: string;
+}
+
+export interface Contributions {
+  [key: string]: {
+    details: ContributionDetails | string;
+    suggestion: Suggestion;
+  };
+}
+
+export interface ReviewTabs {
+  suggestionType: string;
+  text: string;
+}
+
+@Component({
+  selector: 'oppia-contributions-and-review',
+  templateUrl: './contributions-and-review.component.html'
+})
+export class ContributionsAndReview
+  implements OnInit, OnDestroy {
+  directiveSubscriptions = new Subscription();
+
+  SUGGESTION_TYPE_QUESTION: string;
+  SUGGESTION_TYPE_TRANSLATE: string;
+  TAB_TYPE_CONTRIBUTIONS: string;
+  TAB_TYPE_REVIEWS: string;
+  activeExplorationId: string;
+  SUGGESTION_LABELS: SUGGESTION_LABELS;
+  contributions: Contributions;
+  userDetailsLoading: boolean;
+  userIsLoggedIn: boolean;
+  activeTabType: string;
+  activeSuggestionType: string;
+  dropdownShown: boolean;
+  activeDropdownTabChoice: string;
+  reviewTabs: ReviewTabs[] = [];
+  contributionTabs: ContributionTabs[];
+  tabNameToOpportunityFetchFunction: unknown;
+
+  constructor(
+    private alertsService: AlertsService,
+    private contextService: ContextService,
+    private contributionAndReviewService: ContributionAndReviewService,
+    private contributionOpportunitiesService: ContributionOpportunitiesService,
+    private formatRtePreviewPipe: FormatRtePreviewPipe,
+    private ngbModal: NgbModal,
+    private questionObjectFactory: QuestionObjectFactory,
+    private skillBackendApiService: SkillBackendApiService,
+    private translationTopicService: TranslationTopicService,
+    private userService: UserService,
+  ) {}
+
+  getQuestionContributionsSummary(
+      suggestionIdToSuggestions: SuggestionIdToSuggestions):
+      QuestionContributionsSummaryList[] {
+    let questionContributionsSummaryList = [];
```

Comment:
Prefer declaring as `const` if the variable is not being reassigned. Here and elsewhere.

Reply:
done

Reply:
i have taken another look all let has been replaced with const